lkmpg.tex

```diff
@@ -649,30 +649,32 @@ \subsection{Device Drivers}
 
 \section{Character Device drivers}
 \label{sec:orgd3ad4b9}
-\subsection{The proc\_ops Structure}
+\subsection{The file\_operations Structure}
 \label{sec:org3950990}
-The proc\_ops structure is defined in \textbf{/usr/include/linux/fs.h}, and holds pointers to functions defined by the driver that perform various operations on the device. Each field of the structure corresponds to the address of some function defined by the driver to handle a requested operation.
+The file\_operations structure is defined in \textbf{/usr/include/linux/fs.h}, and holds pointers to functions defined by the driver that perform various operations on the device. Each field of the structure corresponds to the address of some function defined by the driver to handle a requested operation.
```

Comment:
Use `\verb|file_operations|` instead.

Reply:
SInce Linux v5.6, the interface has been replaced with `proc_ops`. You should denote.

Reply:
Sorry, do I misunderstand it?
I think `proc_ops` is for files in the VFS, and it is a reduced structure from `file_operations` so normally we still use `file_operations`.

Reply:
Reference [proc: faster open/read/close with "permanent" files](https://lwn.net/Articles/812612/) and quote the key considerations.